opengever/api/task.py

```
@@ -35,6 +42,36 @@ def _get_containing_dossier_summary(self):
             (containing_dossier, self.request), ISerializeToJsonSummary
         )()
 
+    def task_tree(self):
+        main_task = self.context
+        parent = aq_parent(main_task)
+        while ITask.providedBy(parent):
+            main_task = parent
+            parent = aq_parent(main_task)
+
+        solr = getUtility(ISolrSearch)
+        filters = make_filters(
+            path={
+                'query': '/'.join(main_task.getPhysicalPath()),
+                'depth': -1,
+            },
+            object_provides=ITask.__identifier__,
+        )
+        fieldlist = ['Title', 'portal_type', 'path', 'review_state']
+        resp = solr.search(
+            filters=filters, start=0, rows=1000, sort='created asc',
```

can we avoid hardcoding the maximum rows?

---

I understand the sentiment / agree in theory, but there's really no good alternative here I think. Having the frontend deal with paginated responses, just to find out that the DOM becomes unusably slow and impossible to navigate when 1000+ items are rendered doesn't really improve the situation.

---

yeah i was fearing as much, but still i had to ask. what worries me is that we silently hide these tasks. admittedly it may be unlikely to have so many subtasks, though. can we create a follow-up to make sure this is addressed. maybe a hint for the user that there are too many subtasks to display would suffice?

---

I agree that it's not a terribly beautiful solution. But for me this falls into the category 'good enough'. I really do not expect that we will ever have a case that hits this limit. Thus for me handling this, is a waste of time. @deiferni if you disagree, let's make a call to discuss it further.

---

no, i was just wondering/asking if there was an easy way around the hardcoded limit. but it seems to be difficult and deemed unnecessary.
"good enough" works for for me, no further objections 👍